lib/reversetunnel/conn.go

```diff
@@ -251,3 +254,77 @@ func (c *remoteConn) sendDiscoveryRequest(req discoveryRequest) error {
 
 	return nil
 }
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+const emitConnTargetPrefix = "Teleport"
+
+// emitConn is a wrapper for a net.Conn that emits an audit event for non-Teleport connections.
+type emitConn struct {
```

Comment:
Could we move this into its own file? WDYT?

Reply:
I figured that conn.go was an appropriate place to define a conn, but I don't have a strong preference.